e2e/pages/web-pages/home.page.ts

```diff
@@ -0,0 +1,256 @@
+     */
+    async navigateToDevEnvHomepage(): Promise<void> {
+        await this.navigateToUrl(this.devEnvHomepageLink);
+    }
+
+    /**
+     * Navigate to the Prod homepage
+     * NOTE: We could use this method for after deploy smoke tests
+     */
+    async navigateToProdHomepage(): Promise<void> {
+        await this.navigateToUrl(this.prodHomepageLink);
+    }
+
+    /**
+     * Check if Heading is visible by CSS Selector and text with timeout
+     * @param {string} elementSelector
+     * @param {LanguagesEnum} language
+     * @param {string} headingBg
+     * @param {string | null} headingEn
+     */
+    async isHeadingVisibleBySelector(elementSelector: string, language: LanguagesEnum, headingBg: string, headingEn: string | null): Promise<boolean> {
+        await this.waitForElementToBePresentedByLocator(this.page.locator(elementSelector).first());
+        if (language === LanguagesEnum.BG) {
+            return this.isElementVisibleBySelectorWithTimeout(elementSelector, {hasText: headingBg});
+        } else if (language === LanguagesEnum.EN) {
+            return this.isElementVisibleBySelectorWithTimeout(elementSelector, {hasText: headingEn});
+        } else {
+            throw new Error("Language not found!");
+        }
+    }
+
+    /**
+     * Check if H1 heading is visible by text with timeout
+     * @param {LanguagesEnum} language
+     * @param {string} headingBg
+     * @param {string | null} headingEn
+     */
+    async isH1HeadingVisible(language: LanguagesEnum, headingBg: string, headingEn: string | null): Promise<boolean> {
+        return this.isHeadingVisibleBySelector(this.h1HeadingsSelector, language, headingBg, headingEn);
+    }
+ 
+    /**
+     * Check if H4 heading is visible by text with timeout
+     * @param {LanguagesEnum} language
+     * @param {string} headingBg
+     * @param {string | null} headingEn
+     */
+    async isH4HeadingVisible(language: LanguagesEnum, headingBg: string, headingEn: string | null): Promise<boolean> {
+        return this.isHeadingVisibleBySelector(this.h4HeadingsSelector, language, headingBg, headingEn);
+    }
+
+    /**
+     * Check if H5 heading is visible by text with timeout
+     * @param {LanguagesEnum} language
+     * @param {string} headingBg
+     * @param {string | null} headingEn
+     */
+    async isH5HeadingVisible(language: LanguagesEnum, headingBg: string, headingEn: string | null): Promise<boolean> {
+        return this.isHeadingVisibleBySelector(this.h5HeadingsSelector, language, headingBg, headingEn);
+    }
+
+    /**
+     * Check if H6 heading is visible by text with timeout
+     * @param {LanguagesEnum} language
+     * @param {string} headingBg
+     * @param {string | null} headingEn
+     */
+    async isH6HeadingVisible(language: LanguagesEnum, headingBg: string, headingEn: string | null): Promise<boolean> {
+        return this.isHeadingVisibleBySelector(this.h6HeadingsSelector, language, headingBg, headingEn);
+    }
```

Comment:
How and where do you suggest to pass the parameter? 
I think it's better to wrap each method as like now, because it is easier when we have less arguments.